source/server/listener_impl.cc

```diff
@@ -518,6 +518,26 @@ void ListenerImpl::validateFilterChains(Network::Socket::Type socket_type) {
                                        address_->asString()));
     }
   }
+  if (config_.udp_listener_config().has_quic_options()) {
+    for (auto& filter_chain : config_.filter_chains()) {
+      if (!filter_chain.has_transport_socket() ||
+          filter_chain.transport_socket().name() != "envoy.transport_sockets.quic") {
```

Comment:
This will disallow any extensions on quic transport socket. We are already verifying transport socket type in ListenerFilterChainFactoryBuilder::buildFilterChainInternal(). Isn't that not early enough?